pkg/manager/impl/execution_manager.go

```diff
@@ -328,12 +358,28 @@ func (m *ExecutionManager) RelaunchExecution(
 	if executionSpec.Metadata == nil {
 		executionSpec.Metadata = &admin.ExecutionMetadata{}
 	}
+	var inputs *core.LiteralMap
+	if len(existingExecutionModel.UserInputsUri) > 0 {
+		inputs = new(core.LiteralMap)
+		if err := m.storageClient.ReadProtobuf(ctx, existingExecutionModel.InputsUri, inputs); err != nil {
+			return nil, err
+		}
+	} else {
+		// For old data, inputs are held in the spec
```

Comment:
In the rare case that an execution doesn't have inputs, would `spec.Inputs` would be nil or empty here? If nil that may be a problem?